editor/src/clj/editor/resource_watch.clj

```diff
@@ -85,7 +85,7 @@
         lib-states))
 
 (defn- make-builtins-snapshot [workspace]
-  (let [resources (:tree (resource/load-zip-resources workspace (io/resource "builtins.zip")))
+  (let [resources (some->> "builtins.zip" io/resource io/as-file (resource/load-zip-resources workspace) :tree)
```

Comment:
Uhh, I guess I was kind of just rewriting this a bit... Not sure it actually became any clearer from that. 😳

Reply:
it seems like this change the reason of a crash I repo when I try to open a project:
```
2021-12-02 19:05:40.539 16672 [clojure-agent-send-off-pool-4] ERROR editor.dialogs - {:line 424}
java.lang.IllegalArgumentException: Not a file: jar:file:/Users/agulev/Downloads/Defold.app/Contents/Resources/packages/defold-44c7924da32c557c189df250c397d02b72e44d12.jar!/builtins.zip
	at clojure.java.io$fn__11362.invokeStatic(io.clj:61)
	at clojure.java.io$fn__11362.invoke(io.clj:44)
	at clojure.java.io$fn__11336$G__11318__11341.invoke(io.clj:35)
	at editor.resource_watch$make_builtins_snapshot.invokeStatic(resource_watch.clj:88)
	at editor.resource_watch$make_snapshot_info.invokeStatic(resource_watch.clj:193)
	at editor.workspace$make_snapshot_info.invokeStatic(workspace.clj:255)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invoke(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:367)
	at editor.workspace$resource_sync_BANG_.invoke(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:365)
	at editor.boot_open_project$setup_workspace.invokeStatic(boot_open_project.clj:82)
	at editor.boot_open_project$open_project.invokeStatic(boot_open_project.clj:398)
	at editor.boot_open_project$open_project.invoke(boot_open_project.clj:396)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at editor.boot$open_project_with_progress_dialog$fn__90280.invoke(boot.clj:56)
	at editor.dialogs$make_load_project_dialog$fn__25038$fn__25041.invoke(dialogs.clj:422)
	at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
2021-12-02 19:05:40.569 16702 [JavaFX Application Thread] ERROR editor.error-reporting - {:line 110}
java.lang.IllegalArgumentException: Not a file: jar:file:/Users/agulev/Downloads/Defold.app/Contents/Resources/packages/defold-44c7924da32c557c189df250c397d02b72e44d12.jar!/builtins.zip
	at clojure.java.io$fn__11362.invokeStatic(io.clj:61)
	at clojure.java.io$fn__11362.invoke(io.clj:44)
	at clojure.java.io$fn__11336$G__11318__11341.invoke(io.clj:35)
	at editor.resource_watch$make_builtins_snapshot.invokeStatic(resource_watch.clj:88)
	at editor.resource_watch$make_snapshot_info.invokeStatic(resource_watch.clj:193)
	at editor.workspace$make_snapshot_info.invokeStatic(workspace.clj:255)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invoke(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:367)
	at editor.workspace$resource_sync_BANG_.invoke(workspace.clj:363)
	at editor.workspace$resource_sync_BANG_.invokeStatic(workspace.clj:365)
	at editor.boot_open_project$setup_workspace.invokeStatic(boot_open_project.clj:82)
	at editor.boot_open_project$open_project.invokeStatic(boot_open_project.clj:398)
	at editor.boot_open_project$open_project.invoke(boot_open_project.clj:396)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at editor.boot$open_project_with_progress_dialog$fn__90280.invoke(boot.clj:56)
	at editor.dialogs$make_load_project_dialog$fn__25038$fn__25041.invoke(dialogs.clj:422)
	at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

```

Reply:
Ah, it seems this will only happen in a bundled version of the editor. Very good that you thought to check this! ❤️ I've removed the `ZipResource` change from this PR and will create a separate PR for it when I have time to look into the issue.

Reply:
👍thank you! I'll take a look this new build without `ZipResource` change today later.